packages/validator/src/util/validatorRegistrationCache.ts

```diff
@@ -0,0 +1,56 @@
+import {bellatrix} from "@lodestar/types";
+import {PubkeyHex} from "../types.js";
+import {pruneSetToMax} from "./map.js";
+
+/** Maximum number of validators that can connect to a single validator process */
+const MAX_REGISTRATION_IDS = 1_00_000;
+type RegistrationKeyAttributes = {
+  pubKey: PubkeyHex;
+  feeRecipient: string;
+  gasLimit: number;
+};
+
+/**
+ * This cache stores the bellatrix.SignedValidatorRegistrationV1 objects as mev boosts wants
+ * us to send an old regsitration object if there are no changes in the registration data.
+ *
+ * This could potentially be to prevent a DOS attack as every epoch the public builders would have
+ * a hard time processing and verifying the registrations from potentially half a million validators
+ */
+export class ValidatorRegistrationCache {
+  private readonly validatorRegistrationMap = new Map<
+    string,
+    {validatorRegistration: bellatrix.SignedValidatorRegistrationV1; fullKey: string}
```

Comment:
Add alias on type string such that

```ts
type PubkeyHex = string
/** `${feeRecipient}-${gasLimit}` */
type AttributesKey = string
```

Reply:
Since fullKey is nested by first level of pubkey, does it really need to include the pubkey?

Reply:
If you add this cache inside `ValidatorData` you don't need the 2d Map. And then you can prune by number of attribute combinations which is what makes more sense. pubkeys are naturally bounded already